documentation/example_codes/ExampleJobExecutor.m

```diff
@@ -0,0 +1,33 @@
+% Example object which just demonstrates the methods and variables available while parallel
+classdef ExampleJobExecutor < JobExecutor
+    methods
+        function obj = ExampleJobExecutor()
+        % Constructor cannot take args as constructed by JobDispatcher
+        end
+
+        function obj=reduce_data(obj)
+        % Performed at end of do job after synchronise
+        end
+
+        function ok = is_completed(obj)
+        % If returns true, job will not run another cycle of do_job/reduce_data
+        end
+
+        function obj = do_job(obj)
+        % Run once per iteration if "is_completed" does not return true
+
+        % Either an empty array if given int loop count or
+        % data passed through start_job's loop_params
+            obj.loop_data_
+        % Same copy of data initially sent to each process, however,
+        % if changed copy will update on local process
```

Comment:
Does same mean "initially the same as loop data"? Is it a copy of something on the master process? Does the master process also change? Guessing some of this is answered in other documents but it would be convenient to have all of this here in one place.